wds-network-require-login.php

```diff
@@ -229,20 +250,37 @@ public function maybe_auth_redirect() {
 	 * @return null
 	 */
 	public function auth_redirect() {
+		/**
+		 * Filter the whitelist for the requested URL.
+		 *
+		 * @since 0.1.0
+		 *
+		 * @param array $whitelist The array of URLs to whitelist.
+		 */
 		$whitelist   = apply_filters( 'wds_network_require_login_whitelist', array() );
 		$whitelisted = in_array( $this->current_url, $whitelist );
 
+		/**
+		 * Filter the whitelist for the requested path.
+		 *
+		 * @since 0.2.0
+		 *
+		 * @param array $whitelist The array of paths to whitelist.
+		 */
+		$path_whitelist   = apply_filters( 'wds_network_require_login_path_whitelist', array() );
```

Comment:
No, i mean return early/often: 

```php
	public function auth_redirect() {
		$curr_url    = preg_replace( '/\?.*/', '', $this->current_url );
		$login_url   = preg_replace( '/\?.*/', '', wp_login_url() );

		if ( $login_url === $curr_url ) {
			return;
		}

		/**
		 * Filter the whitelist for the requested path.
		 *
		 * @since 0.2.0
		 *
		 * @param array $whitelist The array of paths to whitelist.
		 */
		$path_whitelist   = apply_filters( 'wds_network_require_login_path_whitelist', array() );
		$path_whitelisted = in_array( $this->requested_path, $path_whitelist, true );

		if ( $path_whitelisted ) {
			return;
		}

		/**
		 * Filter the whitelist for the requested URL.
		 *
		 * @since 0.1.0
		 *
		 * @param array $whitelist The array of URLs to whitelist.
		 */
		$whitelist   = apply_filters( 'wds_network_require_login_whitelist', array() );
		$whitelisted = in_array( $this->current_url, $whitelist, true );

		if ( $whitelisted ) {
			return;
		}

		// Ok, do our redirect
		add_filter( 'login_url', array( $this, 'add_our_login_url_filter' ), 10, 3 );
		auth_redirect();
		remove_filter( 'login_url', array( $this, 'add_our_login_url_filter' ), 10 );
	}
```

(I also added the current_url and requested_path as filter parameters)